talk/morelanguage/templates.tex

```diff
@@ -145,27 +146,213 @@
 \end{frame}
 
 \begin{frame}[fragile]
-  \frametitlecpp[98]{Templates}
+  \frametitlecpp[98]{Template specialization}
   \begin{block}{Specialization}
-    templates can be specialized for given values of their parameter
+    Templates can be specialized for given values of their parameter
   \end{block}
+  \small
   \begin{cppcode*}{}
     template<typename F, unsigned int N>
-    struct Polygon {
-      Polygon(F radius) : m_radius(radius) {}
-      F perimeter() {return 2*N*sin(PI/N)*m_radius;}
-      F m_radius;
-    };
+    struct Polygon { ... }; // primary template
 
-    template<typename F>
+    template<typename F> // partial specialization
     struct Polygon<F, 6> {
       Polygon(F radius) : m_radius(radius) {}
-      F perimeter() {return 6*m_radius;}
+      F perimeter() { return 6*m_radius; }
       F m_radius;
     };
+    template<>           // full specialization
+    struct Polygon<int, 6> {
+      Polygon(int radius) : m_radius(radius) {}
+      int perimeter() { return 6*m_radius; }
+      int m_radius;
+    };
+  \end{cppcode*}
+\end{frame}
+
+\begin{advanced}
+
+\begin{frame}[fragile]
+  \frametitlecpp[98]{Template argument deduction}
+  \begin{block}{Template argument deduction}
+    \begin{itemize}
+    \item Only for function templates (Before \cpp14)
+    \item Template arguments deduced from function arguments
```

Comment:
First say what it is, and then mention the restriction?

Reply:
ok!